common/transport/amqp/devdoc/receiver_link_requirements.md

```diff
@@ -0,0 +1,104 @@
+# ReceiverLink Requirements
+
+## Overview
+
+The `ReceiverLink` class is internal to the SDK and shall be considered private. It shall not be used directly by clients of the SDK, who should instead rely on the higher-level constructs (`Client` classes provided by `azure-iot-device` and `azure-iothub`).
+
+The `ReceiverLink` class implements a state machine that manages the underlying `amqp10` link object used to receive messages from IoT Hub. It can be attached and detached manually, and will try to attach automatically if not already attached when setting up a message listener or trying to settle an existing message.
+
+## Example usage
+
+```typescript
+import * as amqp10 from 'amqp10';
+import { AmqpMessage } from './ampq_message';
+
+const linkAddress = 'exampleAddress';
+const amqp10Client = new amqp10.AmqpClient(null);
+
+const receiverLink = new ReceiverLink(linkAddress, null, amqp10Client);
+receiverLink.on('error', (err) => {
+  console.error(err.toString());
+});
+
+receiverLink.on('message', (msg) => {
+  console.log(msg.body);
+  receiverLink.accept(msg, (err) => {
+    if (err) {
+      console.error(err.toString());
+    } else {
+      console.log('message successfully settled (accept)');
+    }
+  });
+});
+```
+
+## Public Interface
+
+### constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient)
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_001: [** The `ReceiverLink` internal state machine shall be initialized in the `detached` state. **]**
```

Comment:
I'll take care of it.

---
In reply to: [129159982](https://github.com/Azure/azure-iot-sdk-node/pull/55#discussion_r129159982) [](ancestors = 129159982,129159537,128120483)